### PR TITLE
[ntsc-1.2] Organize n64dd declarations and convert to 1.0 addresses

### PR DESCRIPTION
**Pull request by cadmic**
Decompollaborate/n64dd uses 1.0 addresses for everything so it's easier if we first convert everything here too. And previously, n64dd declarations were scattered around the codebase, so I put them all into n64dd.h (or another proper palce) and cleaned up some unnecessary undefined syms.

### CODE REVIEW

File: include/segment_symbols.h [hunk 1]
```diff
@@ -84,6 +84,12 @@ DECLARE_SEGMENT(code)
 DECLARE_ROM_SEGMENT(code)
 DECLARE_BSS_SEGMENT(code)
 
+// N64-only, not wrapped in these are not wrapped in an `#if PLATFORM_N64`
+// so that the N64DD code can always be built.
+DECLARE_SEGMENT(n64dd)
+DECLARE_ROM_SEGMENT(n64dd)
+DECLARE_BSS_SEGMENT(n64dd)
```
Comment (Dragorn421) [inline include/segment_symbols.h:91]: I disagree with putting everything into segment_symbols, this header is just as good as variables.h or functions.h

but whatever for now

File: include/n64dd.h [hunk 2]
```diff
@@ -2,39 +2,37 @@
...
+    void (*unk_00)(n64ddStruct_800FEE70_pointers*, struct n64ddStruct_80121220*);
     void (*unk_04)(void);
     void (*unk_08)(struct PlayState* play, struct RoomContext* roomCtx, s32 roomNum);
     void (*unk_0C)(struct PlayState* play);
     void (*unk_10)(struct PlayState* play);
     void (*unk_14)(struct PlayState* play);
     char unk_18[0x1C];
-    void (*unk_34)(struct PauseMapMarksData**);
-    void (*unk_38)(struct PauseMapMarksData**);
+    void (*unk_34)(PauseMapMarksData**);
```
Comment (Dragorn421) [inline include/n64dd.h:34]: why not keep the forward reference

Comment (cadmic) [reply include/n64dd.h]: this is fixing a mistake, `PauseMapMarksData` is not a struct, it's a typedef for an array, and CC_CHECK errors. I broke this in #2117 and fixed it in #2109

Comment (cadmic) [reply include/n64dd.h]: (I've merged main so this diff no longer appears in this PR)

**Review by Dragorn421 — APPROVED** [review]

**Pull request merged**